### PR TITLE
Help Center: Get unread count when new message arrives

### PR DESCRIPTION
**Pull request by escapemanuele**
Related to #

## Proposed Changes

* Update hc unread conversations when:
	- users receive a message and the hc is minimized
	- conversation is opened

## Why are these changes being made?

* We failed to update the unread count in those cases

## Testing Instructions

* Checkout this PR
* Open the HC with `help-center-experience` flag
* Open any unread conversations to mark them as read
* Close the help center and send a message back from zendesk
* Check if the notifications icon is updated accordingly
* Open the help center and the unread conversation
* The icon should change to reflect that all messages have been read

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: packages/help-center/src/components/help-center-smooch.tsx [hunk 1]
```diff
@@ -93,11 +105,17 @@ const HelpCenterSmooch: React.FC< { enableAuth: boolean } > = ( { enableAuth } )
...
+			getUnreadNotifications( allConversations );
 			setZendeskClientId( getClientId( allConversations ) );
+			Smooch.on( 'message:received', getUnreadListener );
 		}
-	}, [ isChatLoaded, setUnreadCount, setZendeskClientId ] );
+
+		return () => {
+			// @ts-expect-error -- 'off' is not part of the def.
+			const unsubscribe = typeof Smooch?.off === 'function' ? Smooch?.off : () => {};
+			unsubscribe( 'message:received', getUnreadListener );
```
Comment (alshakero) [inline packages/help-center/src/components/help-center-smooch.tsx]:
```suggestion
			// @ts-expect-error -- 'off' is not part of the def.
			Smooch?.off?.( 'message:received', getUnreadListener );
```

File: packages/help-center/src/components/utils.tsx [hunk 2]
```diff
@@ -24,7 +24,8 @@ export const getLastMessage = ( { conversation }: { conversation: ZendeskConvers
 };
 
 export const getZendeskConversations = () => {
-	const conversations = Smooch.getConversations();
+	const conversations =
+		typeof Smooch.getConversations === 'function' ? Smooch.getConversations() : [];
```
Comment (alshakero) [inline packages/help-center/src/components/utils.tsx]:
```suggestion
	const conversations =
		typeof Smooch.getConversations?.() ?? [];
```

File: packages/odie-client/src/data/use-get-unread-conversations.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+export const useGetUnreadConversations = () => {
+	const { setUnreadCount } = useDataStoreDispatch( HELP_CENTER_STORE );
+
+	return ( conversations?: Conversation[] | ZendeskConversation[] ) => {
+		const conversationsToCheck = conversations ? conversations : Smooch.getConversations();
+		const { unreadConversations, unreadMessages } = calculateUnread( conversationsToCheck );
+		setUnreadCount( unreadConversations );
+
+		return { unreadConversations, unreadMessages };
+	};
```
Comment (alshakero) [inline packages/odie-client/src/data/use-get-unread-conversations.ts:31]: Tangent: it seems like `setUnreadCount` is entirely not needed, and we can replace it with a resolver for `getUnreadCount`. This way everything will be automatically synced and cached.

**Review by alshakero — COMMENTED** [review]
Just drive by comments.

File: packages/help-center/src/components/utils.tsx (hunk 2 above)
Comment (heavyweight) [reply packages/help-center/src/components/utils.tsx]: replaced with `Smooch?.getConversations?.() ?? []`

**Review by alshakero — COMMENTED** [review]
Code looks good and it works perfectly.

**Review by alshakero — APPROVED** [review]

**Pull request merged**